packages/fx-core/src/common/spec-parser/specParser.ts

```diff
@@ -92,6 +92,23 @@ export class SpecParser {
           content: ConstantString.MultipleServerInformation,
           data: this.spec!.servers,
         });
+      } else if (this.spec!.servers.length === 1) {
+        const serverUrl = this.spec!.servers[0].url;
+        if (serverUrl.startsWith("http://")) {
+          // Http server url is not supported
+          errors.push({
+            type: ErrorType.HttpServerUrlNotSupported,
+            content: ConstantString.HttpServerUrlNotSupported,
+            data: this.spec!.servers,
+          });
+        } else if (!serverUrl.startsWith("https://")) {
```

Comment:
is there any existing path lib to check relative path? If so better to use those apis.